src/smt/theory_str_noodler/decision_procedure.cpp

```diff
@@ -696,199 +705,365 @@ namespace smt::noodler {
         return result;
     }
 
-    LenNode DecisionProcedure::word_to_int(const mata::Word& word, const BasicTerm &var, bool start_with_one, bool handle_invalid_as_from_int) {
-        LenNode result(0);
+    LenNode DecisionProcedure::encode_interval_words(const BasicTerm& var, const std::vector<std::vector<std::pair<mata::Symbol,mata::Symbol>>>& interval_words) {
+        LenNode result(LenFormulaType::OR);
+        for (const auto& interval_word : interval_words) {
+            // How this works on an example:
+            //      interval_word = [4-5][0-9][2-5][0-9][0-9]
```

Comment:
Is there problem with understanding what the function does or how the encoding will look like?
Because the rest of the example gives exactly the encoding (the big disjunct). I changed some words to make it more obvious.

Reply:
I expected something like: for an interval word, the corresponding lia formula is of the form: `x_1*10^4 + x_2*10^3 ...` where `4 <= x_1 <= 5` ... where `x_i` are fresh variables.

Reply:
Aahh, ok; got it